src/clzones.cpp

```diff
@@ -839,17 +839,40 @@ zone_type_id zone_manager::get_near_zone_type_for_item( const item &it,
     }
 
     if( cat.get_id() == item_category_food ) {
-        // skip food without comestible, like MREs
-        if( const item *it_food = it.get_food() ) {
+        const item *it_food = nullptr;
+        bool perishable = false;
+        // Look for food, and whether any contents which will spoil if left out.
+        // Food crafts and food without comestible, like MREs, will fall down to LOOT_FOOD.
+        it.visit_items( [&]( const item * node, const item * parent ) {
```

Comment:
Need targeted capture, capturing the whole environment is an antipattern.
```suggestion
        it.visit_items( [&it_food, &perishable]( const item * node, const item * parent ) {
```